demo/src/main/java/eu/kevin/demo/main/MainViewModel.kt

```diff
@@ -17,71 +28,170 @@ import kotlinx.coroutines.flow.receiveAsFlow
+                    getCreditorsUseCase.getCreditors(countryIso).toListItems()
+                _viewState.update {
+                    it.copy(
+                        creditors = creditors.mapIndexed { index, item ->
+                            item.copy(isSelected = index == 0)
+                        },
+                        loadingCreditors = false
                     )
-                )
-                _viewAction.send(MainViewAction.OpenAccountLinkingSession(state))
-                _viewState.update { MainViewState.Loading(false) }
-            } catch (ignored: Exception) {
-                _viewState.update { MainViewState.Loading(false) }
+                }
+                donationConfiguration.selectedCreditor = creditors.firstOrNull()
+            } catch (e: Exception) {
+                _viewState.update {
+                    it.copy(
+                        loadingCreditors = false,
+                        loadingState = LoadingState.Failure(e)
+                    )
+                }
             }
         }
     }
 
-    fun initializeBankPayment() {
-        viewModelScope.launch(Dispatchers.IO) {
-            _viewState.update { MainViewState.Loading(true) }
-            try {
-                val payment = kevinAuthClient.initializeBankPayment(
-                    InitiatePaymentRequest("0.01")
-                )
-                _viewAction.send(MainViewAction.OpenPaymentSession(payment, PaymentType.BANK))
-                _viewState.update { MainViewState.Loading(false) }
-            } catch (ignored: Exception) {
-                _viewState.update { MainViewState.Loading(false) }
-            }
+    fun onCreditorSelected(creditor: CreditorListItem) {
+        donationConfiguration.selectedCreditor = creditor
+        _viewState.update {
+            it.copy(
+                creditors = it.creditors.map {
+                    it.copy(isSelected = it == donationConfiguration.selectedCreditor)
+                },
+                loadingState = LoadingState.Loading(false)
+            )
+        }
+    }
+
+    fun onPaymentTypeChanged(position: Int) {
+        donationConfiguration.paymentType = PaymentType.values()[position]
+    }
+
+    fun onCountrySelected(iso: String) {
+        _viewState.update {
+            it.copy(
+                selectedCountry = iso,
+                loadingState = LoadingState.Loading(false)
+            )
+        }
+        loadCreditors(iso)
+    }
+
+    fun onSelectCountryClick() {
+        val config = CountrySelectionFragmentConfiguration(
+            _viewState.value.selectedCountry
+        )
+        GlobalRouter.pushModalFragment(CountrySelectionContract.getFragment(config))
+    }
+
+    fun onAmountChanged(amount: String) {
+        _viewState.update {
+            it.copy(
+                buttonText = amount.toRepresentableBigDecimal(),
+                loadingState = LoadingState.Loading(false)
+            )
+        }
+    }
+
+    fun onProceedClick(
```

Comment:
this is not an abstract action, just donate probably would be more correct. Also, supplying payment type from outside (BANK, CARD) maybe also makes sense here because it is selected from the UI. When function accepts more than 2 arguments, it makes sense to wrap those arguments into some object like `DonationRequest` etc.